historydag/dag.py

```diff
@@ -264,13 +293,34 @@ def __init__(self, dagroot: HistoryDagNode, attr: Any = {}):
         assert isinstance(dagroot, UANode)  # for typing
         self.attr = attr
         self.dagroot = dagroot
+        self.current = 0
 
     def __eq__(self, other: object) -> bool:
         # Eventually this can be done by comparing bytestrings, but we need
         # some sorting to be done first, to ensure two dags that represent
         # identical trees return True. TODO
         raise NotImplementedError
 
+    def __getitem__(self, key) -> "HistoryDag":
```

Comment:
Miraculously, this seems to work with negative indices, and actually seems to choose the correct tree! However, I think we should probably handle these explicitly, with something like:
```python
if key < len(self):
    key = len(self) + key
```
Then, _after_, we should require that `key >= 0 and key < len(self)`. For example, `[1,2,3][-5]` raises an `IndexError`.

We should also handle the case in which someone tries to do `dag['some string']` or `dag[1:3]`. In either case, we should raise a `TypeError(f"History DAG indices must be integers, not {type(key)}")` or something.

To test if the passed key is a slice, you can just use `isinstance(key, slice)`.